### ISSUE TITLE
fix: specify dispatcher in network APIs

### ISSUE THREAD

**Issue by Axelen123** [ReVanced/revanced-manager-compose#26]
`ManagerAPI` does not always specify the dispatcher to use.

IO operations should be done in the IO dispatcher and UI updates should be done in the Main dispatcher.

**Comment by CnC-Robert** [comment]
Wouldn't it be better to just get rid of `ManagerAPI`?
You can use `HttpService` for downloading assets and `ReVancedRepository` directly for accessing ReVanced API.

**Comment by Axelen123** [comment]
Yes. I plan on making several changes to the network stack, such as moving to the new API and merging `ManagerAPI` and `ReVancedRepo` into one class to be more inline with the repository pattern.

Download progress tracking will also be refactored, so we will see how that gets dealt with when the time comes.
